packages/gatsby-plugin-image/src/components/gatsby-image.browser.tsx

```diff
@@ -22,7 +23,33 @@ import { Layout } from "../image-utils"
 import { getSizer } from "./layout-wrapper"
 import { propTypes } from "./gatsby-image.server"
 import { Unobserver } from "./intersection-observer"
-import { render } from "react-dom"
+import type { Root } from "react-dom/client"
+
+let reactRender
+if (HAS_REACT_18) {
+  const reactDomClient = require(`react-dom/client`)
+  reactRender = (
+    Component: React.ReactChild | Iterable<React.ReactNode>,
+    el: ReactDOM.Container,
+    root: Root
+  ): unknown => {
+    if (!root) {
+      root = reactDomClient.createRoot(el)
+    }
+
+    root.render(Component)
+
+    return root
```

Comment:
Maybe  I'm missing some context here, but why is with React 18 the reactRender returning something, with older version it's `void`?

Reply:
Because Root doesn't exist in React < 18 and we need the root so we only call createRoot once.